Build/run.sh

```diff
@@ -56,43 +55,36 @@ while [[ $# -gt 0 ]]; do
         ;;
     esac
 done
-if [ "$BRANCH" == "1" ];
-then
-	CONTNAME="$(whoami)-rzg2l_vlp_v3.0.0_$(git branch --show-current)"
-else
-	CONTNAME="$(whoami)-rzg2l_vlp_v3.0.0"
+CONTNAME="$(whoami)-rzg2l_vlp_v3.0.0${BRANCH}"
+if [ ! -z ${VERBOSE} ]; then
+  IGNORE_OUTPUT="2>/dev/null"
 fi
 #Create OUTDIR if it doesn't exist
 if [ ! -d "${OUTDIR}" ];
 then
 	mkdir ${OUTDIR}
 fi
-	if [ -z "${VERBOSE}" ];
-	then
-		chmod 777 ${OUTDIR} 2>/dev/null
-	else
-		chmod 777 ${OUTDIR}
-	fi
+chmod +w ${OUTDIR} ${IGNORE_OUTPUT}
+ret=$?
+if [ $ret -ne 0 ];
+then
+	echo "Unable to obtain full acess  permissions to ${OUTDIR} and its sub directories, edit the permissions of ${OUTDIR} accordingly! exit"
+	exit -1
+fi
 if [ -z "${CPATH}" ]; 
 then
-	/usr/bin/docker run --privileged ${USE_TTY} --rm -e NO=${NO} -e SDK=${SDK} -e DLOAD=${DLOAD} -v "${PWD}/${OUTDIR}":/home/yocto/rzg_vlp_v3.0.0/out --name ${CONTNAME} ${CONTNAME}
+  /usr/bin/docker run --privileged ${USE_TTY} --rm -e NO=${NO} -e SDK=${SDK} -e DLOAD=${DLOAD} -v "${PWD}/${OUTDIR}":/home/yocto/rzg_vlp_v3.0.0/out --name ${CONTNAME} ${CONTNAME}
 else
 	#Create CPATH sub directories if they do not exist
-	if [ ! -d "${CPATH}/downloads" ];
-	then
-		mkdir ${CPATH}/downloads
-	fi
-	if [ ! -d "${CPATH}/sstate-cache/${MPU}" ];
-	then
-		mkdir -p ${CPATH}/sstate-cache/${MPU}
-	fi
-	if [ -z ${VERBOSE} ];
+	mkdir -p ${CPATH}/downloads
+	mkdir -p ${CPATH}/sstate-cache/${MPU}
+
+	chmod -R +w ${CPATH}
+	ret=$?
+	if [ $ret -ne 0 ];
 	then
-		chmod -R 777 ${CPATH}/downloads 2>/dev/null
-		chmod -R 777 ${CPATH}/sstate-cache/${MPU} 2>/dev/null
-	else
-		chmod -R 777 ${CPATH}/downloads
-		chmod -R 777 ${CPATH}/sstate-cache/${MPU}
+		echo "Unable to obtain write permissions to ${CPATH} and its sub directories, edit the permissions of ${CPATH} accordingly! exit"
+		exit -1
 	fi
-	/usr/bin/docker run --privileged ${USE_TTY} --rm -v "${PWD}/${OUTDIR}":/home/yocto/rzg_vlp_v3.0.0/out -v "${CPATH}/downloads":/home/yocto/rzg_vlp_v3.0.0/build/downloads -v "${CPATH}/sstate-cache/${MPU}/":/home/yocto/rzg_vlp_v3.0.0/build/sstate-cache -e NO=${NO} -e SDK=${SDK} -e DLOAD=${DLOAD} --name ${CONTNAME} ${CONTNAME}
+	/usr/bin/docker run --privileged ${USE_TTY} --rm -v "${PWD}/${OUTDIR}":/home/yocto/rzg_vlp_v3.0.0/out -v "${CPATH}/downloads":/home/yocto/rzg_vlp_v3.0.0/build/downloads -v "${CPATH}/sstate-cache":/home/yocto/rzg_vlp_v3.0.0/build/sstate-cache -e NO=${NO} -e SDK=${SDK} -e DLOAD=${DLOAD} --name ${CONTNAME} ${CONTNAME}
```

Comment:
Is there a reason that you removed the `${MPU}` from the cache directory?

Reply:
WIP